geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -1086,6 +1088,32 @@ public static GemFireCacheImpl getForPdx(String reason) {
     clientMetadataService = clientMetadataServiceFactory.apply(this);
   }
 
+  public void lockDiskStore(String diskStoreName) {
+    CountDownLatch countDownLatch = diskStoreLatches.get(diskStoreName);
+    if (countDownLatch == null) {
+      countDownLatch = diskStoreLatches.putIfAbsent(diskStoreName, new CountDownLatch(1));
+      if (countDownLatch != null) {
+        try {
+          countDownLatch.await();
+        } catch (InterruptedException e) {
+          throw new InternalGemFireError(e);
+        }
+      }
+    } else {
+      try {
+        countDownLatch.await();
+      } catch (InterruptedException e) {
+        throw new InternalGemFireError(e);
+      }
+    }
+  }
```

Comment:
The above can be simplified with something like:

```
public void lockDiskStore(String diskStoreName) {
	CountDownLatch latch = JavaWorkarounds.computeIfAbsent(diskStoreLatches, diskStoreName, k -> new CountDownLatch(1));

	try {
		countDownLatch.await();
	} catch (InterruptedException e) {
		throw new InternalGemFireError(e);
	}
}
```

Reply:
In your simplified code, all thread will `await()`, which is not correct.